src/components/TodoList/TodoList.tsx

```diff
@@ -1,12 +1,78 @@
 /* eslint-disable */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../app/store';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { actions } from '../../features/todos';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { getTodos } from '../../api';
+type Props = {
+  loader: boolean;
+  setLoader: (loader: boolean) => void;
+  setModal: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const TodoList: React.FC<Props> = ({ loader, setLoader, setModal }) => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const dispatch = useDispatch();
+  const todos: Todo[] = useSelector((state: RootState) => state.todos);
+  const add = (value: Todo) => dispatch(actions.addTodo(value));
+
+  const status = useSelector<RootState>(state => state.filter.status);
+  const text = useSelector<RootState, string>(state => state.filter.text);
+  const { id: currentId } =
+    useSelector((state: RootState) => state.currentTodo) || {};
+
+  const handliClick = (todo: Todo) => {
+    dispatch(currentTodoActions.setTodo(todo));
+    setModal(true);
+  };
+
+  useEffect(() => {
+    if (visibleTodos) {
+      setLoader(false);
+    }
+  }, [visibleTodos]);
+
+  useEffect(() => {
+    getTodos().then((todos: Todo[]) => {
+      console.log(todos);
+
+      todos.forEach(todo => add(todo));
+    });
```

Comment:
looks strange, better to create some action in todo reducer that will handle adding all todos in one action